node_test.go

```diff
@@ -122,5 +123,43 @@ func TestSomeNodesHandshake(t *testing.T) {
 	log.SetFlags(fl)
 	log.SetOutput(os.Stderr)
 	log.Print(out)
+}
+
+// go test -benchmem -run=^$ -benchmem -memprofile memprofile.out -cpuprofile cpuprofile.out -bench=BenchmarkHandshakeProfile
+// go tool pprof {file}
+func BenchmarkHandshakeProfile(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		var peersNumber int = 1
+		nodeASocket := "127.0.0.1:9000"
+		var peers []*Node
+
+		configurationA := config.New()
+		configurationA.Write(config.SetSelfListeningAddress(nodeASocket))
+		nodeA := New(configurationA)
 
+		for i := 0; i < peersNumber; i++ {
+			port := 9001 + i
+			address := fmt.Sprintf("127.0.0.1:%v", port)
+
+			configuration := config.New()
+			configuration.Write(config.SetSelfListeningAddress(address))
+			node := New(configuration)
+			peers = append(peers, node)
+		}
+
+		fmt.Println("********************** Listen **********************")
+		// TODO: When node listen to a closed port throws a panic
```

Comment:
Seems like we return an error to avoid panic, https://github.com/geolffreym/p2p-noise/blob/4e1b89c8ff4fd4a3396818cb9899505a375062ff/node.go#L219 in our case maybe we can run a func routine with the listen function inside to handle the error?

Reply:
According to effective go “ But what if the error is unrecoverable? Sometimes the program simply cannot continue.” we can use panic in this case, if we run many routines with listen, panic will stop the others , so we need to implement an extra recover to avoid it..

Reply:
Something that we can approach to make the listen “more testable” it’s use in the SelfListeningAddress a “nil” or empty value to force the listener to find a valid port https://pkg.go.dev/net#Listen